pandas/core/arrays/arrow/accessors.py

```diff
@@ -0,0 +1,187 @@
+"""Accessors for arrow-backed data."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pandas.compat import pa_version_under7p0
+
+if not pa_version_under7p0:
+    import pyarrow as pa
+    import pyarrow.compute as pc
+
+    from pandas.core.dtypes.dtypes import ArrowDtype
+
+if TYPE_CHECKING:
+    from pandas import (
+        DataFrame,
+        Series,
+    )
+
+
+class StructAccessor:
+    """
+    Accessor object for structured data properties of the Series values.
+
+    Parameters
+    ----------
+    data : Series
+        Series containing Arrow struct data.
+    """
+
+    _validation_msg = "Can only use the '.struct' accessor with 'struct[pyarrow]' data."
+
+    def __init__(self, data=None) -> None:
+        self._parent = data
+        self._validate(data)
+
+    def _validate(self, data):
+        dtype = data.dtype
+        if not isinstance(dtype, ArrowDtype):
+            raise AttributeError(self._validation_msg)
```

Comment:
IMO I think `TypeError` would be more appropriate here.

Also could we include the incoming type in the error message like `"f..., not {dtype}"`

Reply:
I think AttributeError was likely preventing this test failure.

```
pandas/tests/series/test_api.py:175: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../micromamba/envs/test/lib/python3.9/inspect.py:351: in getmembers
    value = getattr(object, key)
pandas/core/accessor.py:224: in __get__
    accessor_obj = self._accessor(obj)
pandas/core/arrays/arrow/accessors.py:38: in __init__
    self._validate(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pandas.core.arrays.arrow.accessors.StructAccessor object at 0x7ff513c9cbe0>
data = Series([], dtype: object)

    def _validate(self, data):
        dtype = data.dtype
        if not isinstance(dtype, ArrowDtype):
>           raise TypeError(self._validation_msg.format(dtype=dtype))
E           TypeError: Can only use the '.struct' accessor with 'struct[pyarrow]' dtype, not object.

pandas/core/arrays/arrow/accessors.py:43: TypeError
```